metrics/testutils/testutils.go

```diff
@@ -17,19 +17,33 @@ type ExpectedMetric struct {
 
 // TODO do something similar for Timers
 
-// AssertCounterMetrics checks if counter metrics exist.
+// AssertCounterMetric checks if counter metric exists.
+func AssertCounterMetric(t *testing.T, f *metrics.LocalFactory, expectedMetrics ...ExpectedMetric) {
+	counters, _ := f.Snapshot()
+	assertMetrics(t, counters, expectedMetrics...)
+}
+
+// AssertGaugeMetric checks if gauge metric exists.
+func AssertGaugeMetric(t *testing.T, f *metrics.LocalFactory, expectedMetrics ...ExpectedMetric) {
+	_, gauges := f.Snapshot()
+	assertMetrics(t, gauges, expectedMetrics...)
+}
+
+// AssertCounterMetrics checks if the existing counter metrics exactly match the expected metrics.
 func AssertCounterMetrics(t *testing.T, f *metrics.LocalFactory, expectedMetrics []ExpectedMetric) {
```

Comment:
this function checks the size as well, ie exact match versus do metrics exist in the set